src/sparse/linalg/mod.rs

```diff
@@ -9,7 +9,8 @@ pub mod etree;
 mod ordering;
 pub mod trisolve;
 
-pub use self::ordering::cuthill_mckee;
+pub use self::ordering::reverse_cuthill_mckee;
+pub use self::ordering::cuthill_mckee_custom;
```

Comment:
As you mentionned:

> Concerning the namespacing, that definitely needs to be overhauled. I introduced the modules start and order, but they need to be publicly re-exported in some sensible way or something. It's up to you to decide if and how to do that. Also, maybe a different file structure is needed.

without re-exporting the traits and their implementors `cuthil_mckee_custom` is not callable. I'm starting to think the best solution could be to make the `ordering` module public, keeping only the `pub use self::ordering::reverse_cuthill_mckee;` re-export to guide towards the default algorithm.

Reply:
Will do that as described.